pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -165,6 +165,9 @@ func (r *ReconcileClusterDeployment) Reconcile(request reconcile.Request) (recon
 	}
 
 	if cd.DeletionTimestamp != nil {
+		if cd.Spec.PreserveOnDelete {
+			cdLog.Warn("skipping deprovisioning due to PreserveOnDelete")
+		}
```

Comment:
We're double warning here, lets remove this block and we'll warn below.

Reply:
done.